### PR TITLE
Add more logs in the progress check path

### PR DESCRIPTION
**Pull request by 0xmichalis**
Follow-up to https://github.com/kubernetes/kubernetes/pull/41097

@mfojtik quick look here

### CODE REVIEW

File: pkg/controller/deployment/deployment_controller.go [hunk 1]
```diff
@@ -796,5 +797,6 @@ func (dc *DeploymentController) checkForProgress(key string) (bool, error) {
 	if err != nil {
 		return false, err
 	}
+	glog.V(2).Infof("Syncing deployment %q for a progress check", key)
```
Comment (mfojtik) [inline pkg/controller/deployment/deployment_controller.go]: how often this message apprear in the log?

Comment (kargakis) [inline pkg/controller/deployment/deployment_controller.go]: As often as "Queueing up deployment..." in progress.go. That shouldn't be that often - depends on the # of pods and it's v2. We can raise the lvl later, now I want them purely for debugging an ongoing flake that is blocking the merge queue (https://github.com/kubernetes/kubernetes/issues/39785).

Comment (kargakis) [inline pkg/controller/deployment/deployment_controller.go]: https://github.com/kubernetes/kubernetes/issues/39785 flakes only on aws and gke and thank god I can get back controller manager logs in v2 from aws.

**Comment by deads2k** [comment]
no functional changes, logging only, retest not required

**Comment by kargakis** [comment]
I can't believe that the test this PR was about has two days to flake in aws...

**Pull request merged**